osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -71,8 +75,8 @@ protected override double StrainValueOf(DifficultyHitObject current)
         }
 
         protected override double GetPeakStrain(double offset)
-            => applyDecay(individualStrain, offset - Previous[0].BaseObject.StartTime, individual_decay_base)
-               + applyDecay(overallStrain, offset - Previous[0].BaseObject.StartTime, overall_decay_base);
+            => applyDecay(individualStrain, offset - Previous.First().BaseObject.StartTime, individual_decay_base)
```

Comment:
pretty sure all these `.First()` calls are going to allocate enumerators, which we want to avoid. i think you're looking for `.Peek()`

Reply:
Ah good point, will make the change in the morning :+1:

Reply:
Fixed in 7e0cdb250c9008140ba363760ac6e031718e8d74

Reply:
Fixed in 7e0cdb250c9008140ba363760ac6e031718e8d74